modules/determining-upgrade-viability-conditiontype.adoc

```diff
@@ -0,0 +1,38 @@
+// Module included in the following assemblies:
+//
+// * updating/index.adoc
+
+:_content-type: CONCEPT
+[id="Understanding-upgrade-viability-conditiontype_{context}"]
+= Understanding upgrade viability with conditionType
+
+The status of cluster operators is revealed with the value of conditionType, informing you of potential impacts to upgrading your cluster. 
+This information can be retrieved in either YAML or JSON format through the `oc get clusterversion -o` command:
+
+[source,terminal]
+----
+$ oc get clusterversion -o YAML
+----
+
+* Available: 
+The conditionType `Available` indicates that the operators are functional and available in the cluster. If the status is `Available=False`, then the status indicates that at least one part of the operand is non-functional and the condition requires an administrator to intervene.
+
+* Progressing:
+The conditionType `Progressing` indicates that the operator is actively rolling out new code, propagating configuration changes, or otherwise moving from one steady state to another. Operators should not report as `Progressing` when they are reconciling a previous known state. If the observed cluster state has changed and the operator is reacting to it, then the status should report back as `Progressing=True`, since it is moving from one steady state to another.
+
+* Degraded:
+A `Degraded` conditionType indicates that the operator's current state does not match its desired state over a period of time. The period of time may vary by component, but a `Degraded` state represents persistent observation of a condition.  As a result, a component should not fluctuate in and out of the `Degraded` state.  
++
+A service may have the conditionType `Available`, even if it is degraded. This indicates that the service is `Available` but `Degraded` because it may have a lower quality of service. 
++
+There may be a different conditionType if the transition from one state to another does not persist over a long enough period to report `Degraded`.  
+A service should not report `Degraded` during the course of a normal upgrade.  A service may report `Degraded` in response to a persistent infrastructure failure that requires eventual administrator intervention.
```

Comment:
This sentence seems to be (operator) developer-oriented rather than administrator oriented.  Also, we do not speak about "services" in the previous descriptions, just in this one. We should probably use consistent terminology.

Reply:
+1 to "operator" or "component" instead of "service" here.  [The API Godocs][1] this paragraph is descended from use "component".

[1]: https://github.com/openshift/api/blob/cce310ad2932f6de24491052d506926e484c082c/config/v1/types_cluster_operator.go#L171-L173